_CS141/type-level-programming.md

```diff
@@ -157,16 +157,92 @@ stwo :: SNat ('Succ ('Succ 'Zero))
 stwo :: SSucc sone
 ```
 
-## Proxy Types
+## Reification
 
-We don’t have types at runtime due to **type erasure**, so we can’t write a function that does something based on a type. 
+> Reification is the process of **converting types to values**, typically using **type classes**. This resolves to the correct instances at compile-time so the right implementation of the function is evaluated at runtime.
```

Comment:
I find the second sentence a bit unnatural here, because it is a detail that I think makes a lot more sense _after_ the explanation of how reification works.

Reply:
Moved it to a conclusion subsection.